src/app/components/ad/ad.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit, Input } from '@angular/core'
+import { DomSanitizer, SafeHtml } from '@angular/platform-browser'
+import { ModalController } from '@ionic/angular'
+import { SubscribePage } from 'src/app/pages/subscribe/subscribe.page'
+import AdUtils, { AdLocation } from 'src/app/utils/AdUtils'
+import { MerchantUtils } from 'src/app/utils/MerchantUtils'
+
+@Component({
+	selector: 'app-ad',
+	templateUrl: './ad.component.html',
+	styleUrls: ['./ad.component.scss'],
+})
+export class AdComponent implements OnInit {
+	info: AdLocation = 'info'
+
+	@Input() location: AdLocation
+
+	adHtml: SafeHtml
+	openUpgradeToPremium = false
+
+	isAdFree = false
+
+	constructor(
+		private adUtils: AdUtils,
+		private sanitizer: DomSanitizer,
+		private modalController: ModalController,
+		private merchantUtils: MerchantUtils
+	) {}
+
+	ngOnInit() {
+		this.merchantUtils.hasAdFree().then((isAdFree) => {
+			this.isAdFree = isAdFree
+		})
+		this.adUtils.get(this.location).then((data) => {
```

Comment:
Minimal: You could skip this call for when `this.isAdfree == true`.
I know that this is checked within `adUtils.get` too, but checking it here would be more performant.

Feel free to ignore and to resolve this conversation.

Reply:
The first thing hasAdFree does (or to be more precise the underlying getCurrentPlanConfirmed method) is checking if the user is signed in and if not returning the default package. So there isn't really any performance gain by checking here as well.